aiohttp/client_reqrep.py

```diff
@@ -537,12 +537,15 @@ async def write_bytes(
 
             await writer.write_eof()
         except OSError as exc:
-            new_exc = ClientOSError(
-                exc.errno, "Can not write request body for %s" % self.url
-            )
-            new_exc.__context__ = exc
-            new_exc.__cause__ = exc
-            protocol.set_exception(new_exc)
+            if exc.errno is None and isinstance(exc, asyncio.TimeoutError):
```

Comment:
If this is a pattern that's going to be repeated, it should probably be put into a pure function...